lib/appenders/file.js

```diff
@@ -65,7 +69,12 @@ function fileAppender (file, layout, logSize, numBackups, compress, timezoneOffs
   openFiles.push(logFile);
 
   return function(loggingEvent) {
-    logFile.write(layout(loggingEvent, timezoneOffset) + eol, "utf8");
+    if (paused) {
```

Comment:
Wouldn't it be more elegant to write:
 
    if (!paused && !logFile.write(layout(loggingEvent, timezoneOffset) + eol, "utf8")) {
      process.stdout.write('write buffer full (possibly due to errors), dropping logs until it recovers.\n');
      paused = true;
    }
?